libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts

```diff
@@ -79,62 +147,195 @@ export class TableVirtualScrollDirective extends TableVirtualScroll implements O
+                { passive: false }
+            );
+        } else {
+            this._tableScrollDispatcher
+                .verticallyScrolled()
+                .pipe(
+                    filter(() => this.virtualScroll && !!this.bodyHeight),
+                    takeUntil(this._destroy$)
+                )
+                .subscribe((scrollable) => {
+                    this.calculateVirtualScrollRows();
+                    this._table.setTableState({
+                        ...this._table.getTableState(),
+                        scrollTopPosition: scrollable.getScrollTop()
+                    });
+                });
+        }
+    }
+
+    /** @hidden */
+    private _getNumberOfRowsToDisplay(): number {
+        const tableContainer = this._table.tableContainer.nativeElement;
+        let tableHeight = tableContainer.clientHeight;
+        tableHeight = tableHeight - tableContainer.querySelector('thead').clientHeight;
+        return Math.floor(tableHeight / (this.rowHeight + 2));
+    }
+
+    /** @hidden */
+    private _wheelScrollListenerFunction = (event: WheelEvent): void => {
+        if (Math.abs(event.deltaX) < Math.abs(event.deltaY)) {
+            event.preventDefault();
+            event.stopImmediatePropagation();
+            if (this._wheelTimeout) {
+                window.cancelAnimationFrame(this._wheelTimeout);
+            }
+            const deltaY = event.deltaY;
+            if (deltaY) {
+                this._wheelTimeout = window.requestAnimationFrame(() => {
+                    this._scrollRow(deltaY > 0 ? 1 : -1);
+                });
+            }
+        }
+    };
+
+    /** @hidden */
+    private _mockScrollbarListenerFunction = (event: Event): void => {
+        event.preventDefault();
+        event.stopImmediatePropagation();
+        if (this._scrollMockTimeout) {
+            window.cancelAnimationFrame(this._scrollMockTimeout);
+        }
+        const tableScrollMockContainer = this._table.tableScrollMockContainer.nativeElement;
+        let deltaY = tableScrollMockContainer.scrollTop - this._lastMockScrollPosition;
+        deltaY = Math.ceil(deltaY / this.rowHeight);
+        this._scrollMockTimeout = window.requestAnimationFrame(() => {
+            this._scrollRow(deltaY, true);
+            this._lastMockScrollPosition = tableScrollMockContainer.scrollTop;
+            if (tableScrollMockContainer.scrollTop === 0) {
+                this._setRows(0);
+            }
+            if (
+                tableScrollMockContainer.scrollTop + tableScrollMockContainer.getBoundingClientRect().height ===
+                tableScrollMockContainer.scrollHeight
+            ) {
+                const startingNodeIndex = this._table._tableRowsVisible.length - this._getNumberOfRowsToDisplay();
+                this._setRows(startingNodeIndex);
+            }
+        });
+    };
+
+    /** @hidden */
+    private _scrollRow(count: number, fromMockScrollbarListener = false): void {
```

Comment:
it's little bit non understandable how this method is working, maybe you should add some description before *hidden